core/lib/via_withdrawal_client/src/client.rs

```diff
@@ -0,0 +1,148 @@
+use std::{collections::HashMap, str::FromStr};
+
+use zksync_da_client::DataAvailabilityClient;
+use zksync_types::{web3::keccak256, H160, H256};
+
+use crate::{
+    pubdata::Pubdata,
+    types::{L2BridgeLogMetadata, WithdrawalRequest, L2_BASE_TOKEN_SYSTEM_CONTRACT_ADDR},
+    withdraw::parse_l2_withdrawal_message,
+};
+
+#[derive(Debug)]
+pub struct WithdrawalClient {
+    client: Box<dyn DataAvailabilityClient>,
+}
+
+impl WithdrawalClient {
+    pub fn new(client: Box<dyn DataAvailabilityClient>) -> Self {
+        Self { client }
+    }
+
+    pub async fn get_withdrawals(&self, blob_id: &str) -> anyhow::Result<Vec<WithdrawalRequest>> {
+        let pubdata_bytes = self._fetch_pubdata(blob_id).await?;
+        let pubdata = Pubdata::decode_pubdata(pubdata_bytes)?;
+        let l2_bridge_metadata = WithdrawalClient::_list_l2_bridge_metadata(&pubdata);
+        let withdrawals = WithdrawalClient::_get_valid_withdrawals(l2_bridge_metadata);
+        Ok(withdrawals)
+    }
+
+    async fn _fetch_pubdata(&self, blob_id: &str) -> anyhow::Result<Vec<u8>> {
```

Comment:
maybe we can keep our previous rust-convention and don't use leading underscore for private functions? it seems weird for me, but if you insist, that's ok
https://doc.rust-lang.org/1.0.0/style/style/naming/README.html

Reply:
Fixed